src/dotnet-scaffolding/dotnet-scaffold-aspnet/ScaffoldSteps/AddFileStep.cs

```diff
@@ -0,0 +1,54 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using Microsoft.DotNet.Scaffolding.Core.Scaffolders;
+using Microsoft.DotNet.Scaffolding.Core.Steps;
+using Microsoft.DotNet.Scaffolding.Internal.Services;
+using Microsoft.DotNet.Scaffolding.TextTemplating;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.Tools.Scaffold.AspNet.ScaffoldSteps;
+
+internal class AddFileStep : ScaffoldStep
+{
+    //this file name should be found in dotnet-scaffold-aspnet\Templates\Files
+    public required string FileName { get; set; }
+    public required string BaseOutputDirectory { get; set; }
+    private readonly ILogger _logger;
+    private readonly IFileSystem _fileSystem;
+
+    public AddFileStep(ILogger<AddFileStep> logger, IFileSystem fileSystem)
+    {
+        _logger = logger;
+        _fileSystem = fileSystem;
+    }
+
+    public override Task<bool> ExecuteAsync(ScaffolderContext context, CancellationToken cancellationToken = default)
+    {
+        var destinationFilePath = Path.Combine(BaseOutputDirectory, FileName);
+        if (string.IsNullOrEmpty(BaseOutputDirectory) ||
```

Comment:
Should not first check `BaseOutputDirectory` and `FileName` are not null/empty before attempting to construct the `destinationFilePath`?

From documentation, `Path.Combine` throws `ArgumentNullException` if any of the params is null.

Reply:
oh good to know, will add a check before the Path.Combine